File: src/EPI_Java/epi/Problem_5_1.java
package epi;

/**
 * Created by rick on 2017/1/16.
 */
public final class Problem_5_1 {
    public static boolean parity(long value) {
        boolean result = false;
        while (value != 0){
           value &= value - 1;
           result = !result;
        }
        return result;
    }
}

File: src/EPI/EPI_CSharp/Problem_5_1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EPI_CSharp
{
    public static class Problem_5_1
    {
        /// <summary>
        /// parity of a integer
        /// </summary>
        /// <param name="value"></param>
        /// <returns>true if value is an odd integer, false otherwise</returns>
        public static bool Parity(long value)
        {
            var uValue = (ulong)value;
            var result = false;
            while (uValue != 0)
            {
                uValue &= uValue - 1;
                result = !result;
            }
            return result;
        }
    }
}
